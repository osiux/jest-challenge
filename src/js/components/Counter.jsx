import React from "react";
import PropTypes from "prop-types";

const Counter = ({ counter }) => (
    <div className="card-body">
        <p className="counter" style={{ fontSize: "500%" }}>
            {counter}
        </p>
    </div>
);

Counter.propTypes = {
    counter: PropTypes.number.isRequired
};

export default Counter;
