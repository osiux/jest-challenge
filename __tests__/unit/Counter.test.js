import React from 'react';
import { shallow } from 'enzyme';

import Counter from '../../src/js/components/Counter';

describe('Counter component', () => {
    it('should match snapshot', () => {
        const component = shallow(<Counter counter={0} />);

        expect(component).toMatchSnapshot();
    });

    it('should render counter from property', () => {
        const component = shallow(<Counter counter={10} />);

        expect(component.find('p.counter').text()).toBe('10');
    });
});
