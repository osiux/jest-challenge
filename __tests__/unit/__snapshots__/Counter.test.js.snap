// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should match snapshot 1`] = `
<div
  className="card-body"
>
  <p
    className="counter"
    style={
      Object {
        "fontSize": "500%",
      }
    }
  >
    0
  </p>
</div>
`;
